File: frontend/src/types/report.ts
export interface ReportVO {
  id: number
  title: string
  description: string
  contentMd: string
  contentHtml: string
  chartConfig: Record<string, any>
  sessionId: string
  status: number
  version: number
  createdBy: string
  createdAt: string
  updatedAt: string
}

export interface ReportVersionVO {
  id: number
  reportId: number
  version: number
  contentMd: string
  changeNote: string
  createdBy: string
  createdAt: string
}

File: backend/src/main/java/com/dataagent/report/entity/Report.java
package com.dataagent.report.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@TableName("report")
public class Report {
    @TableId(type = IdType.AUTO)
    private Long id;

    private String title;
    private String description;
    private String contentMd;
    private String contentHtml;
    private String chartConfig;
    private String dataJson;
    private String sessionId;
    private Integer status;
    private Integer version;
    private Long createdBy;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
}
